ui-react/packages/atlasmap-ui/src/AtlasmapUI/AtlasmapCanvasView/AtlasmapCanvasView.tsx

```diff
@@ -47,12 +55,26 @@ export const AtlasmapCanvasView: FunctionComponent<IAtlasmapCanvasViewProps> = (
     showTypes,
   ]);
   const [showMappingPreview, setShowMappingPreview] = useState(false);
+
   const toggleShowMappingPreview = useCallback(() => {
     const newValue = !showMappingPreview;
     setShowMappingPreview(newValue);
     onShowMappingPreview(newValue);
   }, [onShowMappingPreview, showMappingPreview]);
 
+  const conditionalMappingExpressionEnabled = useCallback(() => {
+    return onConditionalMappingExpressionEnabled();
+  }, [onConditionalMappingExpressionEnabled]);
+
+  const getMappingExpressionStr = useCallback(() => {
+    return onGetMappingExpressionStr();
+  }, [onGetMappingExpressionStr]);
+
+  const toggleExpressionMode = useCallback(() => {
+    onToggleExpressionMode();
+    expressionTokens[0] = 'ratsky';
```

Comment:
Same comment as above about setting and resetting variables inside event handlers.